nimbus/src/lib.rs

```diff
@@ -319,6 +319,7 @@ pub struct Experiment {
     pub bucket_config: BucketConfig,
     pub probe_sets: Vec<String>,
     pub branches: Vec<Branch>,
+    #[serde(default)]
```

Comment:
```rust
// ⚠️ Warning : Altering this type might require a DB migration. ⚠️
```

This change affects the reading of experiments that are serialized and are missing a `featureIds` field.

The `Experiment` objects in the database either have a `feature_ids` property, in which case, this line has no effect; or not: in which case we default to an empty array.

In this PR we test that this deserialization with and without a `featureIds` field. 

I believe that we do not need to do any migration at this point.

Reply:
> I believe that we do not need to do any migration at this point.

+1, your reasoning makes sense to me w.r.t. not needing a db migration.

Longer-term, we could consider adding some explicit tests for reading data in order formats from the persistence layer, which I think is pretty similar to what you're doing here, but structured in a way that makes it more obvious *why* it's important to test these things. The [fxa-client state-persistence tests](https://github.com/mozilla/application-services/blob/3eda75aa501fad5161d8c0538c7b3cdf89da18b9/components/fxa-client/src/state_persistence.rs#L236) might be good inspiration here. (Each of these tests as a comment saying "It's really important that you don't modify the test data here, as that would defeat the point of the test" precisely because we were accidentally putting up PRs that modified the b/w-compat test data :-/)